spec/end_to_end_spec.rb

```diff
@@ -0,0 +1,162 @@
+require "open4"
+require "socket"
+
+describe "EndToEnd" do
+  COLOR_REGEX = /\e\[(\d+)(;\d+)*m/
+  PATH_REGEX  = /^\/.+\n/
+  AFTER_RUNNING_REGEX = /running after_job_finishes/
+  HOSTNAME = Socket.gethostname
+
+  before(:all) do
+    Dir.chdir(File.join(__dir__, "..", "tests", "end_to_end")) do
+      pid, stdin, stdout, stderr = Open4::popen4("bundle exec gorgon")
+      @outputs = stdout.read.split("*" * 80)
+    end
+  end
+
+  context "number of output hunks" do
+    it "is same as errors raised plus meta information" do
+      expect(@outputs.count).to eq(7), "expected 7 output hunks, got:\n#{@outputs.inspect}"
+    end
+  end
+
+  context "meta suite run information" do
+    it "has version and legend information" do
+        expected_meta = <<-META
+Welcome to Gorgon #{Gorgon::VERSION}
+before job starts was called
+Loading environment and workers...\r                                  \rLegend:
+F - failure files count
+H - number of hosts that have run files
+W - number of workers running files
+
+
+Progress: ||
+
+        META
```

Comment:
Does `rspec` have some sort of snapshot testing so we don't have to maintain these strings? 👀  @arturopie